BosTau.py

```diff
@@ -0,0 +1,215 @@
+"""
+Catalog definitions for Bos Taurus
+"""
+import math
+import logging
+
+import msprime
+
+import stdpopsim
+
+logger = logging.getLogger(__name__)
+
+###########################################################
+#
+# Genome definition
+#
+###########################################################
+
+
+# Create a string of chromosome lengths for easy parsing
+# Length data was extracted from Ensembl genome browser Eg. for Chr1 ....https://www.ensembl.org/Bos_taurus/Location/Chromosome?r=1:421347-158534110
+_chromosome_data = """\
+chr1    158534110
+chr2    136231102
+chr3    121005158
+chr4    120000601
+chr5    120089316
+chr6    117806340
+chr7    110682743
+chr8    113319770
+chr9    105454467
+chr10   103308737
+chr11   106982474
+chr12   87216183
+chr13   83472345
+chr14   82403003
+chr15   85007780
+chr16   81013979
+chr17   73167244
+chr18   65820629
+chr19   63449741
+chr20   71974595
+chr21   69862954
+chr22   60773035
+chr23   52498615
+chr24   62317253
+chr25   42350435
+chr26   51992305
+chr27   45612108
+chr28   45940150
+chr29   51098607
+chrX    139009144
+"""
+# Parse list of chromosomes into a list of Chromosome objects which contain the
+# chromosome name, length, mutation rate, and recombination rate
+_chromosomes = []
+
+for line in _chromosome_data.splitlines():
+    name, length = line.split()[:2]
+    _chromosomes.append(stdpopsim.Chromosome(
+        id=name, length=int(length),
+        mutation_rate=1.2e-8,
+        recombination_rate=9.26e-9)) # 25.5 crossovers per meiosis in males and 23.2 crossovers per meiosis in females gives an average of 24.35 crossovers per meiosis
```

Comment:
Mutation and recombination rates look good! Relatively easy to see where you got them in the papers